build.sbt

```diff
@@ -20,24 +19,40 @@ startYear := Some(2019)
 organizationName := "Evolution Gaming"
 
 organizationHomepage := Some(url("http://evolutiongaming.com"))
+coverageExcludedFiles := ".*CacheOpsCompat.*"
+coverageEnabled := crossSettings(
+  scalaVersion.value,
+  if3 = false,
+  if2 = true
+)
 
 scalaVersion := crossScalaVersions.value.head
 
-crossScalaVersions := Seq("2.13.11", /*"3.2.0", */"2.12.18")
+crossScalaVersions := Seq("2.13.11", "3.3.0", "2.12.18")
+
+scalacOptsFailOnWarn := crossSettings(
+  scalaVersion.value,
+  if3 = Some(false),
```

Comment:
turned off for Scala 3 as there are a lot of warnings for pattern matches `the type test for com.evolution.scache.LoadingCache.EntryState.Value[F,
  com.evolution.scache.ExpiringCache.Entry[V]] cannot be checked at runtime...`

Reply:
switched it to `coverallsFailBuildOnError`

Reply:
Is that a compiler bug or does it indeed has a chance of error? I wouldn't want to miss some possible errors in Scala 3 build accidentally 🤔

Reply:
It's because of covariance in `sealed trait EntryState[+F[_], +A]`

Making it covariant in `final case class Value[+F[_], +A](entry: Entry[F, A]) extends EntryState[F, A]` will help, but it doesn't work with `final case class Loading[F[_], A](deferred: Deferred[F, Either[Throwable, Entry[F, A]]]) extends EntryState[F, A]` because of invariant `Deferred`.